src/components/validated_input.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div>
+    <input id="input"
+           :class="{error : errors.length > 0}"
+           type="text"
+           :value="input_value"
+           @input="$e => _change_input($e.target.value)"/>
+    <h4 id="invalid-input-header"
+        v-if="!is_valid()"
+        class="error-text">Invalid input!</h4>
+    <ul>
+      <li v-for="error of errors"
+          class="error-message error-text">{{error}}</li>
+    </ul>
+  </div>
+</template>
+
+<script lang="ts">
+  import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+
+  type ValidatorResponse = [boolean, string];
+  type ValidatorFuncType = (value: string) => ValidatorResponse;
+  type FromStringFuncType = (value: string) => unknown;
+  type ToStringFuncType = (value: unknown) => string;
+
+  function default_to_string_func(value: unknown): string {
+    return "" + value;
+  }
+
+  function default_from_string_func(value: string): unknown {
+    return value;
+  }
+
+  @Component
+  export default class ValidatedInput extends Vue {
+    @Prop({required: true})
+    value!: unknown;
+
+    @Prop({required: true, type: Array})
+    validators!: ValidatorFuncType[];
+
+    @Prop({required: false})
+    to_string_fn!: ToStringFuncType;
+
+    @Prop({required: false})
+    from_string_fn!: FromStringFuncType;
+
+    input_value: string = "";
+    errors: string[] = [];
+
+    private _to_string_fn(value: unknown): string {
+      return (this.to_string_fn !== undefined) ? this.to_string_fn(value)
+        : default_to_string_func(value);
+    }
+
+    private _from_string_fn(value: string): unknown {
+      return (this.from_string_fn !== undefined) ? this.from_string_fn(value)
+        : default_from_string_func(value);
+    }
+
+    // Note: This assumes "value" is valid when passed as prop to this component
+    mounted() {
```

Comment:
Would there be any issue with running the validators here? Seems like doing so would be more robust.